gateway/cert.go

```diff
@@ -448,6 +481,15 @@ func (gw *Gateway) getTLSConfigForClient(baseConfig *tls.Config, listenPort int)
 
 		// Cache the config
 		tlsConfigCache.Set(hello.ServerName+listenPortStr, newConfig, cache.DefaultExpiration)
```

Comment:
@buger shouldn't we set the cache after we do the `gwConfig.HttpServerOptions.SkipClientCAAnnouncement` validation?